articles/how-to-fix-and-prevent-sql-injection-in-wordpress/index.md

```diff
@@ -0,0 +1,194 @@
+![wordpress-core-files](/engineering-education/how-to-fix-and-prevent-sql-injection-in-wordpress/wordpress-core-files.png)
+Ask yourself, is WordPress Core running on the updated patch?
+
+As we have found out, your WordPress site is secure from any SQLi if your core files are updated. Nevertheless, use of third-party plugins and themes change the narrative and exposes you exponentially. We shall list the most prone WordPress plugin installation areas that you be on the look-out. They include;
+
+1. Subscription pop-ups
+2. Login/Signup forms
+3. E-commerce checkout pages/carts
+4. Search bars
+5. Contact forms
+6. Feedback forms
+7. Generic contact forms
+8. Blog comment forms
+9. Search parameters
+
+The [Exploit-DB website](https://www.exploit-db.com/search?q=) provides a detailed context of vulnerabilities happening in WordPress plugins. Furthermore, it points out how the attacks affected victim websites and the solution deployed.  
+
+### How to Detect and Fix WordPress SQLi Attack?
+
+##### Fix WordPress SQLi Attack: Damage Control
+
+An attacker targets the database at any given point as that is the only source for information on your website. Firstly, disable admin privileges on your database. Henceforth, the attacker is tied down a read-only database that is not of much help to them.
+
+Proceed to sanitize the UDF files the attacker used to obtain the OS Shell. In addition, change your encrypted password to more stronger values. Finally, check if the port 3306 is open on the internet and if so, block it.
+
+##### Fix WordPress SQLi Attack: Database Clean Up
+
+A database clean-up is required to fix any possible SQLi on your site's database. Here are steps to so as well with the function cheat-sheet.
+
+Step 1: Look up all your tables using the command show tables; Look out for a table named Sqlmap.
+
+Step 2: If the table Sqlmap is present, it has been used against your website. Delete the Sqlmap table using the command 
+~~~~sql
+DROP TABLE Sqlmap;
+~~~~
+
+Step 3: Look out for new/foreign users in your db using the following code
+
+~~~~sql
+Select * FROM users  AS u AND u.created > UNIX_TIMESTAMP(STR_TO_DATE('sept 22 2020', '%M %d %Y '));
+~~~~
+Manipulate the string in the code to determine when any of the users found were created.
+
+Step 4: Delete any rogue users using the command below:
+
+~~~~sql
+DROP USER 'malicious'@'localhost';
+~~~~
+
+Step 5: Change and secure your passwords using the command below.
+~~~~sql
+UPDATE users SET pass = concat('ZZZ', sha(concat(pass, md5(rand()))));
+~~~~
+
+##### Fix WordPress SQLi Attack: Vital updates
+
+Updates for your WordPress site are very vital in keeping ahead of day to day vulnerabilities. Make sure to update your WordPress Core files oftenly as well as your theme and plugins. Avoid using nulled themes on your WP site. In addition, make sure to use reputable plugins on your site.
```

Comment:
often not oftenly
You could link to an explanation of nulled themes. Also, there are nulled plugins too.

Reply:
I have linked the text for both nulled themes and plugins.